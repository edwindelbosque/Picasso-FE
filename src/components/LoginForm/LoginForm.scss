.Form {
	margin-top: 300px;
}
