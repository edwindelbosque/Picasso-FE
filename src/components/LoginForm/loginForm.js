import React, { useState } from "react";
// import userLogin from '../../util/apiCalls.js'

const LoginForm = (props) => {
    console.log(props);
    
    const [emailValue, handleEmailChange] = useState("");
    const [passwordValue, handlePassowrdChange] = useState("");

    const handleSubmit = async event => {
        event.preventDefault();
        const newUser = {email: emailValue, passowrd: passwordValue}
        const loginResponse = await userLogin(newUser)
        if(loginResponse) {
            console.log(loginResponse);
            // userName = loginResponse
            // userId = loginResponse
            resetInputs();
        } else {

        }
      }

    const resetInputs = () => {
        handleEmailChange('')
        handlePassowrdChange('')
    }

    const checkEmail = () => {
         const emailSplit = emailValue.split('')
         const startIncludesSearch = emailSplit.findIndex(character => character === '@')
         return emailSplit.includes('@') && emailSplit.includes('.', startIncludesSearch) ? true : false
       }

    const canBeSubmitted = () => {
          return (checkEmail() && passwordValue.length > 6 )
        }

    const isEnabled = canBeSubmitted()

    return (
        <form className='Form'>
            <label htmlFor="email" className="form-login email-login__label">
                Email
            </label>
            <input
                className="emailForm"
                id="email"
                type="text"
                name="email"
                placeholder="Email"
                onChange={ e => handleEmailChange(e.target.value)}
                value={emailValue}
            />
             <label htmlFor="password" className="form-login passowrd-login__label">
                Password
            </label>
            <input
                className="emailForm"
                id="password"
                type="password"
                name="password"
                placeholder="Password"
                onChange={ e => handlePassowrdChange(e.target.value)}
                value={passwordValue}
            />
            <button
                type="button"
                className="loginFormBtn"
                disabled={!isEnabled}
                onClick={(e) => handleSubmit(e)}
                >
                Submit
            </button>
       </form>
    )
}

export default LoginForm;