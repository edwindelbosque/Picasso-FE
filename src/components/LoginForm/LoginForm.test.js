import React from 'react';
import LoginForm from './LoginForm';
import { shallow } from 'enzyme';
jest.mock('react-redux', () => ({
	useDispatch: () => {}
}));

describe('LoginForm', () => {
	let wrapper;
	const updateCurrentUser = jest.fn();
	const toggleMenu = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
			<LoginForm
				updateCurrentUser={updateCurrentUser}
				toggleMenu={toggleMenu}
			/>
		);
	});

	it('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
