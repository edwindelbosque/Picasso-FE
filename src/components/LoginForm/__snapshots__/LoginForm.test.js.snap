// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm App should match snapshot 1`] = `
<form
  className="Form"
>
  <label
    className="form-login email-login__label"
    htmlFor="email"
  >
    Email
  </label>
  <input
    className="emailForm"
    id="email"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <label
    className="form-login password-login__label"
    htmlFor="password"
  >
    Password
  </label>
  <input
    className="emailForm"
    id="password"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <button
    className="loginFormBtn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</form>
`;
